xla/service/gpu/fusions/transpose_mlir.cc

```diff
@@ -214,6 +212,9 @@ MlirTransposeFusion::WriteResult MlirTransposeFusion::EmitWriteToShMemMlir(
     ValueRange output_args) const {
   std::vector<int64_t> shmem_tensor_size(tiling_.GetBlockTileSize().begin(),
                                          tiling_.GetBlockTileSize().end());
+  shmem_tensor_size.erase(shmem_tensor_size.begin() +
+                          (permutation_ == Vector3({0, 2, 1}) ? 0 : 1));
```

Comment:
Can we just keep this? The 1 doesn't really hurt, and this line is not trivial to understand without context.

Reply:
If we keep this, we should make [the logic](https://github.com/openxla/xla/pull/12328/files#diff-0210cd152294b1475469517ffba6b5ee578acc4e786b0dceeee6854aba445a29L190) here more complex, because c0 should place on different position. Which do you recommend?

Reply:
I'd prefer that, if it's not too much logic. FYI we also have some plans to extend this emitter to support arbitrary transposes (not just ones that can be mapped to 3d).